src/realm/sync/transform.cpp

```diff
@@ -1833,6 +1844,10 @@ DEFINE_NESTED_MERGE(Instruction::ArrayMove)
 DEFINE_MERGE(Instruction::ArrayMove, Instruction::ArrayMove)
 {
     if (same_container(left, right)) {
+        REALM_MERGE_ASSERT(left.prior_size == right.prior_size);
+        REALM_MERGE_ASSERT(left.index() <= left.prior_size);
+        REALM_MERGE_ASSERT(right.index() <= right.prior_size);
```

Comment:
Should we check both indexes here? The from and the to index of the move? Also, dont these also need to be `<` as opposed to `<=`